### ISSUE TITLE
[reconfigurator] Add garbage collection for expunged datasets

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/omicron#6646]
Related to https://github.com/oxidecomputer/omicron/issues/5552

https://github.com/oxidecomputer/omicron/pull/6229 adds dataset to the blueprint, to make their existence explicit instead of implicit alongside zones and disks.

Like zones, we are initially implementing "add and expungement" of datasets initially, with removal being deferred until we're confident we can observe the removal of datasets from the inventory.

**Comment by davepacheco** [comment]
I assume that in the same way that #5552 is blocked on being sure that the expunged zone is gone from inventory, this issue is blocked on being sure that the expunged dataset is gone from inventory, which in turn depends on #7304.

**Comment by smklein** [comment]
This was discussed on https://github.com/oxidecomputer/omicron/issues/7319 - if we opt  to not merge https://github.com/oxidecomputer/omicron/pull/7325 , then "removing expunged datasets" should depend on bundles with expunged datasets being reassigned explicitly
